_posts/2022-08-05-hacking-for-cruise-control-ui.md

```diff
@@ -0,0 +1,300 @@
+        - name: cluster-ca-certs
+          mountPath: /etc/ssl/certs/
+      volumes:
+        - name: cruise-control-ui
+          configMap:
+            name: cruise-control-ui
+        - name: api-auth-config
+          secret:
+            defaultMode: 292
+            secretName: my-cluster-cruise-control-api
+        - name: cluster-ca-certs
+          secret:
+            defaultMode: 292
+            secretName: my-cluster-cruise-control-certs
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cruise-control-ui
+  labels:
+    app: cruise-control-ui
+data:
+  default.conf: |2+
+        server {
+        listen 9090 ssl;
+        server_name localhost; 
+        ssl_certificate     /etc/ssl/certs/cruise-control.crt;
+        ssl_certificate_key /etc/ssl/certs/cruise-control.key;
+       
+        location / {
+            root   /usr/share/nginx/html;
+            index  index.html index.htm;
+        }
+
+        location /kafkacruisecontrol/ {
+            proxy_set_header Host $host;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_buffering off;
+            proxy_pass https://my-cluster-cruise-control:9090/kafkacruisecontrol/;
+            proxy_ssl_verify              off;
+        }
+        
+        error_page   500 502 503 504  /50x.html;
+        location = /50x.html {
+            root   /usr/share/nginx/html;
+        }
+        }
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: cruise-control-ui
+  labels:
+    app: cruise-control-ui
+spec:
+  selector:
+    app: cruise-control-ui
+  ports:
+    - protocol: TCP
+      port: 9090
+      targetPort: 9090
+```
+
+Then run it:
+```
+kubectl apply -f cruise-control-ui.yaml -n <namespace>
+```
+
+## Using Cruise Control Front End
+
+After being deployed, the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) application will be accessible via the cruise-control-ui `service`.
+
+If running the cluster locally with [minikube](https://github.com/kubernetes/minikube), you can port-forward the Cruise Control UI `service` to access it locally:
+
+```
+kubectl port-forward svc/cruise-control-ui 9090:9090
+```
+
+Then navigate [‘https://127.0.0.1:9090’](https://127.0.0.1:9090) in your browser
+
+![Cruise Control Frontend](/assets/images/posts/2019-08-08-cruise-control-frontend.png)
+
+## Conclusion
+
+Although it is possible to run a [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) instance alongside a Strimzi-managed Kafka cluster it is only possible when the Cruise Control REST API security settings are disabled.
+Having these security settings disabled exposes the Cruise Control API to potentially destructive Cruise Control operations that are not coordinated or supported by Strimzi Operator.
+If using the Cruise Control UI with a Strimzi-managed Kafka cluster, it is best to stick with using it solely for monitoring your cluster, not for operating on it.
+That being said, use at your own risk!
```

Comment:
We have this in the intro now. Do we still want it in the conclusion too? I think having these caveats is better in the intro. Then we could reserve the conclusion to what we've achieved by going ahead: got a Cruise Control UI up-and-running -- which is great for monitoring.